lib/src/commands/chat_command.dart

```diff
@@ -471,6 +358,13 @@ class ChatCommand
       throw CommandRegistrationError('Invalid localized name for command "$name".');
     }
 
+    Type contextType = IChatContext;
+    if (resolvedOptions.type == CommandType.slashOnly) {
+      contextType = InteractionChatContext;
+    } else {
+      contextType = MessageChatContext;
+    }
```

Comment:
ternary could be used there. Also first assignemt doesnt make sense since you set it to MessageChatContext in else anyway.

```suggestion
    Type contextType = resolvedOptions.type == CommandType.slashOnly
       ? InteractionChatContext
       : MessageChatContext;
```